### DISCUSSION TITLE
Is Rack Attack useful against DDOS attacks?

### DISCUSSION THREAD

**Discussion by OZcomingFRoo** [kickstarter/rack-attack#659] [Questions (Q&A)]
In my company, we implamented Rack Attack against brute force attacks.

I would like to use Rack attack in my own personal project for both Brute force and DDOS attacks.

The official guide does not mention DDOS, but it seems like the gem should be useful for such purpose

**Comment by mpenalozag** [comment]
@OZcomingFRoo I don't think you can protect from DDoS only by using `Rack::Attack`. The defense this gems provides is at the application layer. If lots of requests get to your app, they are consuming resources and threads from your webserver either way. For DDoS protection you would need setting up a service like Cloudflare or, for example, setting up a Load Balancer in GCP and protect it with Cloud Armor. Other cloud providers have similar systems.
